azkaban-common/src/test/java/azkaban/user/XmlUserManagerTest.java

```diff
@@ -73,6 +80,90 @@ public void testDoNotExist() throws Exception {
     fail("XmlUserManager should throw an exception when the file doesn't exist");
   }
 
+  /**
+   * Test auto reload of user XML
+   */
+  @Test
+  public void testAutoReload() throws Exception {
+    final Props props = new Props(this.baseProps);
+    final URL configURL = Resources.getResource("test-conf/azkaban-users-test1.xml");
+    final String path = configURL.getPath();
+    props.put(XmlUserManager.XML_FILE_PARAM, path);
+
+    UserManager manager;
+    try {
+      manager = new XmlUserManager(props);
+    } catch (final RuntimeException e) {
+      fail("Should have found the xml file");
+      return;
+    }
+
+    // Get the user8 from existing XML with password == password8
+    User user8 = manager.getUser("user8", "password8");
+
+    // Modify the password for user8
+    // TODO : djaiswal : Find a better way to modify XML
+    List<String> lines = new ArrayList<>();
+    List<String> origLines = new ArrayList<>();
+    Path filePath = Paths.get(path);
+    FileTime origModifiedTime = Files.getLastModifiedTime(filePath);
+    System.out.println("File modification time = " + origModifiedTime.toString());
+    for (String line : Files.readAllLines(filePath)) {
+      origLines.add(line);
+      if (line.contains("password8")) {
+        lines.add(line.replace("password8", "passwordModified"));
+      } else {
+        lines.add(line);
+      }
+    }
+
+    // Make sure the file gets reverted back.
+    try {
+      // Update the file
+      Files.write(filePath, lines);
+      FileTime lastModifiedTime = Files.getLastModifiedTime(filePath);
+      System.out.println("File modification time after write = " + lastModifiedTime.toString());
+      if (origModifiedTime.equals(lastModifiedTime)) {
+        // File did not update
+        fail("File did not update.");
+      }
+      // Try for a minute polling every 2 seconds if the config reloaded
+      try {
```

Comment:
have you considered https://github.com/awaitility/awaitility/wiki/Usage ?

Reply:
Thanks for the tip. I got it working with this library. Patch coming in.